pkg/kubelet/config/config.go

```diff
@@ -325,6 +323,20 @@ func filterInvalidPods(pods []*api.Pod, source string, recorder record.EventReco
 	return
 }
 
+// checkAndUpdatePod updates existing if ref makes a meaningful change and returns true, or
+// returns false if there was no update.
+// TODO: it would be better to update the whole object and only preserve certain things
+//       like the source annotation or the UID (to ensure safety)
+func checkAndUpdatePod(existing, ref *api.Pod) bool {
```

Comment:
The workaround for #8642 was to explicitly set all default fields in the static pods, so you'll need to update fluentd with the default terminationGracePeriod as was done for cpu in #8657.